db_store/src/metric_tracker.rs

```diff
@@ -0,0 +1,41 @@
+use std::time::Duration;
+
+use crate::{Error, Result};
+
+const DURATION: Duration = Duration::from_secs(300);
+
+pub async fn start(
+    app_name: String,
+    pool: sqlx::Pool<sqlx::Postgres>,
+    shutdown: triggered::Listener,
+) -> Result<futures::future::BoxFuture<'static, Result>> {
+    let join_handle = tokio::spawn(async move { run(app_name, pool, shutdown).await });
+
+    Ok(Box::pin(async move {
+        match join_handle.await {
+            Ok(()) => Ok(()),
+            Err(err) => Err(Error::from(err)),
+        }
+    }))
+}
+
+async fn run(app_name: String, pool: sqlx::Pool<sqlx::Postgres>, shutdown: triggered::Listener) {
```

Comment:
Since you format the string, you don't really need to own it _I think_ (it's not across an await boundary, but there could be another issue), so you can probably just make this &str. Then you can get rid of the `impl Into<String>` and make that `&str` as well. Hopefully I'm right and this doesn't cause you to spend time on something that ends up not compiling

Reply:
Address in 574c46a